Added the possibility to give a FormatError handler

lucapiccinelli opened the pull request:
First of all, congratulations for the wonderful work you did on OpenPop.

I would like to suggest this little modification: it happened to me that a mail massage stored on disk contained bad mail headers or bad content-type specification. It follows that while reading these fault contents, OpenPop throws a FormatError stopping the processing on the message remnant.

With my modification you can choose to pass a IParsingErrorHandler to Message.Load or to Pop3Client.GetMessage in order to handle the error on the specific broken header but without stopping message reading. As a result you can obtain the Message you where trying to read/get even if some of its content is broken, receiving notification of the broken content. If you don't give any handler (leaving this parameter to null) it throws the eventual exception, just like before the modification.

Look forward for your answer, have a nice day!

Thank you,
Luca

foens commented:
Hi there and sorry for not giving fast feedback.
I would gladly pull in your merge request, however, you removed `<AssemblyOriginatorKeyFile>../OpenPopKeyFile.pfx</AssemblyOriginatorKeyFile>` from the .csproj files. If you add them again, I will merge.

lucapiccinelli commented:
Thank you, no problem about the not so fast feedback :). I removed the key file as i was not able to compile without having the OpenPopKeyFile.pfx. I reverted those changes and reintegrated the key as your request... but again i wasn't able to compile it anymore as i still don't have any OpenPopKeyFile.pfx.

Thank you, have a nice day
Luca
